src/useViewEncapsulationRule.ts

```diff
@@ -0,0 +1,47 @@
+import { getComponentDecorator, getDecoratorPropertyInitializer } from './util/utils';
+import { ComponentMetadata } from './angular/metadata';
+import * as Lint from 'tslint';
+import * as ts from 'typescript';
+
+import { NgWalker } from './angular/ngWalker';
+
+export class Rule extends Lint.Rules.AbstractRule {
+
+  static metadata: Lint.IRuleMetadata = {
+    ruleName: 'use-view-encapsulation',
+    type: 'maintainability',
+    description: 'Disallows using of ViewEncapsulation.None',
+    rationale: '',
+    options: null,
+    optionsDescription: 'Not configurable',
+    typescriptOnly: true
+  };
+
+  static FAILURE = 'Using "ViewEncapsulation.None" may cause conflicts between css rules having the same selector';

```

Comment:
Thank you. Please refer to 88e3cd417204b063e1bd2155c29de3bd7a2132fa to see the updated warning.